source/credentials.c

```diff
@@ -199,8 +207,17 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
 
     environment_provider = aws_credentials_provider_new_environment(allocator, &environment_options);
 
+    bool success = true;
+    struct aws_string *ecs_relative_uri = NULL;
+    aws_get_environment_value(allocator, s_ecs_creds_env_relative_uri, &ecs_relative_uri);
+    struct aws_string *ecs_full_uri = NULL;
+    aws_get_environment_value(allocator, s_ecs_creds_env_full_uri, &ecs_full_uri);
+    struct aws_string *ec2_imds_disable = NULL;
+    aws_get_environment_value(allocator, s_ec2_creds_env_disable, &ec2_imds_disable);
```

Comment:
need to check return code of `aws_get_environment_value()`

Reply:
Thought of in the following code, testing if the string pointers are `NULL` and if the `len` vars are non zero show cover the return of this function.

Reply:
If `aws_get_environment_value()` fails, it's because the user **does** have the environment variable set, but we ran out of memory and couldn't return the string. So we'd be doing the wrong thing to proceed as if the environment variable wasn't set

Reply:
Got your point, I'll change this.